Signal/src/ViewControllers/SelectRecipientViewController.m

```diff
@@ -0,0 +1,553 @@
+                    return;
+                }
+
+                NSString *recipientId = recipients[0].uniqueId;
+                [activityAlert dismissViewControllerAnimated:NO
+                                                  completion:^{
+                                                      [weakSelf.delegate phoneNumberWasSelected:recipientId];
+                                                  }];
+            }
+            failure:^(NSError *error) {
+                OWSAssert([NSThread isMainThread]);
+                if (wasCancelled) {
+                    return;
+                }
+                [activityAlert dismissViewControllerAnimated:NO
+                                                  completion:^{
+                                                      [ViewControllerUtils
+                                                          showAlertWithTitle:NSLocalizedString(@"ALERT_ERROR_TITLE",
+                                                                                 @"Title for a generic error alert.")
+                                                                     message:error.localizedDescription];
+                                                  }];
+            }];
+    } else {
+        NSString *recipientId = possiblePhoneNumbers[0];
+        [self.delegate phoneNumberWasSelected:recipientId];
+    }
+}
+
+- (void)textFieldDidChange:(id)sender
+{
+    [self updatePhoneNumberButtonEnabling];
+}
+
+// TODO: We could also do this in registration view.
+- (BOOL)hasValidPhoneNumber
+{
+    if (!self.callingCode) {
+        return NO;
+    }
+    NSString *possiblePhoneNumber =
+        [self.callingCode stringByAppendingString:self.phoneNumberTextField.text.digitsOnly];
+    PhoneNumber *parsedPhoneNumber = [PhoneNumber tryParsePhoneNumberFromUserSpecifiedText:possiblePhoneNumber];
+    // It'd be nice to use [PhoneNumber isValid] but it always returns false for some countries
+    // (like afghanistan) and there doesn't seem to be a good way to determine beforehand
+    // which countries it can validate for without forking libPhoneNumber.
+    return parsedPhoneNumber && parsedPhoneNumber.toE164.length > 1;
+}
+
+- (void)updatePhoneNumberButtonEnabling
+{
+    BOOL isEnabled = [self hasValidPhoneNumber];
+    self.phoneNumberButton.enabled = isEnabled;
+    [self.phoneNumberButton
+        setBackgroundColor:(isEnabled ? [UIColor ows_signalBrandBlueColor] : [UIColor lightGrayColor])];
+}
+
+#pragma mark - CountryCodeViewControllerDelegate
+
+- (void)countryCodeViewController:(CountryCodeViewController *)vc
+             didSelectCountryCode:(NSString *)countryCode
+                      countryName:(NSString *)countryName
+                      callingCode:(NSString *)callingCode
+{
+
+    [self updateCountryWithName:countryName callingCode:callingCode countryCode:countryCode];
+
+    [self textField:self.phoneNumberTextField shouldChangeCharactersInRange:NSMakeRange(0, 0) replacementString:@""];
```

Comment:
Yes, I should have added a comment.  The intent is to trigger the formatting logic with a no-op edit, since changing the country code potentially changes the formatting behavior. I've added a comment.